### PR TITLE
fix(formula-editor): handle unary-prefixed variables in `buildRPN`

### PR DESCRIPTION
**Pull request by paras-fundwave**
#### Description

Formula editor exposes a parser and a validator, out of which validator does allow for unary operators but the parser return a "Invalid Formula" error.

This PR fixes that and also added test-cases for formula editor.

#### Resolutions

- resolves failing test-cases on main - [seen here](https://github.com/fundwave/fw-components/actions/runs/28103589053/job/83211701574#step:9:34)

#### Depends on

N/A

#### Deployment

Current pipelines will suffice

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved unary plus/minus handling across calculation, RPN generation, parentheses formatting, and precedence.
  * Enhanced input validation errors for unknown variables, consecutive operators, trailing operators, and unmatched parentheses.
  * Division by zero now consistently returns no result (`undefined`).

* **New Features**
  * Added comprehensive parser tests covering arithmetic, exponentiation, unary edge cases, precedence, and parentheses formatting.

* **Chores**
  * Bumped the package version and updated the formula-editor package entrypoints and test scripts.

### CODE REVIEW

File: packages/formula-editor/package.json [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
-  "main": "dist/formula-editor/src/formula-editor.js",
+  "main": "dist/formula-editor.js",
   "exports": {
-    ".": "./dist/formula-editor/src/formula-editor.js",
-    "./types": "./dist/formula-editor/src/types",
-    "./types/*.js": "./dist/formula-editor/src/types/*.js",
-    "./utils/*.js": "./dist/formula-editor/src/utils/*.js"
+    ".": "./dist/formula-editor.js",
+    "./utils/*.js": "./dist/utils/*.js"
   },
```
[the remark this answers is not in the document]
Comment (paras-fundwave) [reply packages/formula-editor/package.json:9]: we will release this as minor bump - breaking changes as it's not best practice to expose types like that

File: packages/formula-editor/test/parser.test.ts [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+  ]);
+  const parser = new Parser(vars, 0);
+
+  it("does not add redundant parentheses when precedence is unambiguous", () => {
+    const result = parser.addParentheses("A + B * C");
+    expect(result).toBe("A + B * C");
+  });
+
+  it("adds parentheses when lower-precedence op is nested", () => {
+    const result = parser.addParentheses("(A + B) * C");
```
Comment (parasdeep03) [inline packages/formula-editor/test/parser.test.ts]: result and argument are same ?

Comment (parasdeep03) [reply packages/formula-editor/test/parser.test.ts]: can make the argument a bit complex like 
```
(A + B) * C / A

**Review by parasdeep03 — DISMISSED** [review]
Reviewed the fixes
- common method to check operands
- replacing redundant if checks with the common method
- test cases for parser

**Review by parasdeep03 — APPROVED** [review]
